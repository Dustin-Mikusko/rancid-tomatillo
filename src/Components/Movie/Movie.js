import React from 'react';
import './Movie.scss';
import { Link } from 'react-router-dom';
import RatingForm from '../../Containers/RatingForm/RatingForm';
import PropTypes from 'prop-types';

const Movie = ({id, title, releaseDate, poster, backdrop, overview, avgRating, user }) => {
  let userRating;
  if (user && user.ratings.find(rating => rating.movie_id === id)) {
    userRating = user.ratings.find(rating => rating.movie_id === id);
  }

  releaseDate = releaseDate.split("-").reverse().join("/").slice(3)

  return (
    <article className='movie-card'>
      <img src={poster} className='movie-poster'/>
      <section className='all-card-info'>
        <div className='title-rating-wrap'>
          <div className='title-date'>
            <h2 className='movie-card-title'>{title}</h2>
            <p className='release-date'>Released: {releaseDate}</p>
          </div>
          <div className='ratings-section'>
            <p className='avg-rating'>Avg Rating: {avgRating.toFixed()}</p>
            {userRating && <p className='your-rating'> Your Rating: {userRating.rating}</p>}
            {user && <RatingForm userId={user.id} ratingId={userRating ? userRating.id : undefined} movieId={id}/>}
          </div>
        </div>
        <Link to={`movies/${id}`} className='show-more'>Show More</Link>
      </section>
    </article>
  )
}

export default Movie;

Movie.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  releaseDate: PropTypes.string,
  poster: PropTypes.string,
  backdrop: PropTypes.string,
  overview: PropTypes.string,
  avgRating: PropTypes.number,
  user: PropTypes.object 
}
