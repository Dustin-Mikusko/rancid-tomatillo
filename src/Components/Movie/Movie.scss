@import url('https://fonts.googleapis.com/css?family=Quicksand:500,700&display=swap');

.movie-card {
  margin: 25px auto ;
  width: 55%;
  background-color: rgba(0, 0, 0, .6);
  border: 4px ridge white;
  height: 300px;
  color: #FFF;
  display: flex;
  border-radius: 3px;
  font-family: 'Quicksand', sans-serif;
  font-weight: 700;
}

.movie-poster {
  width: 31%;
  padding: 5px;
}

.all-card-info {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}

.title-rating-wrap {
  display: flex;
  width: 100%;
}

.title-date {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 15px;
  width: 49%;
}

.movie-card-title {
  font-size: 170%;
}

.release-date {
  font-size: 134%;
}

.your-rating,
.avg-rating {
  font-size: 134%;
}

.ratings-section {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  margin: 20px 0;
  width: 51%;
}

.show-more {
  text-decoration: none;
  color: #FFF;
  font-size: 150%;
  margin-bottom: 15px;
  border: 2px ridge #FFF;
  background: #000;
  border-radius: 3px;
  width: 50%;
    &:hover {
      font-style: italic;
    }
    &:active {
      transform: scale(.97)
    }
}