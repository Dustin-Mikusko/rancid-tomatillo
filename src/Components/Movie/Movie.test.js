import React from 'react';
import Movie from './Movie';
import { shallow } from 'enzyme';

describe('Movie', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Movie />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there is a userRating', () => {
    const mockUser = {
      ratings: [
        { name: 'Knives Out', movie_id: 10 },
        { name: 'Moana', movie_id: 7 }
      ]
    };
    const mockId = 7;

    const wrapper = shallow(<Movie user={mockUser} id={mockId}/>);

    expect(wrapper).toMatchSnapshot();
  })
});
