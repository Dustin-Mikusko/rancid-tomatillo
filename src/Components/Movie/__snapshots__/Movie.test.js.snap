// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Movie should match the snapshot when there is a userRating 1`] = `ShallowWrapper {}`;
